pkg/controller/component/lifecycle/kbagent.go

```diff
@@ -252,32 +267,37 @@ func (a *kbagent) templateVarsParameters() (map[string]string, error) {
 	return m, nil
 }
 
-func (a *kbagent) callActionWithSelector(ctx context.Context, spec *appsv1alpha1.Action, lfa lifecycleAction, req *proto.ActionRequest) error {
+func (a *kbagent) callActionWithSelector(ctx context.Context, spec *appsv1alpha1.Action, lfa lifecycleAction, req *proto.ActionRequest) ([]byte, error) {
 	pods, err := a.selectTargetPods(spec)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if len(pods) == 0 {
-		return fmt.Errorf("no available pod to call action %s", lfa.name())
+		return nil, fmt.Errorf("no available pod to call action %s", lfa.name())
 	}
 
 	// TODO: impl
 	//  - back-off to retry
 	//  - timeout
+	var output []byte
```

Comment:
would both stdout and stderr be captured? 

seems should be separated into to two []byte

Reply:
I see, the output is the http response..

Reply:
The data of stderr is carried by the message of error.